Sources/Apodini/Configurations/HTTPConfiguration.swift

```diff
@@ -8,75 +8,96 @@
 
 import Foundation
 import NIO
+import NIOSSL
 
-/// A `Configuration` for HTTP.
-/// The configuration can be done in two ways, either via the
-/// command line arguments --hostname, --port and --bind or via the
-/// function `address`
+/// A `Configuration` for HTTP, HTTP/2 and TLS.
+///
+/// Examples of config via code:
+/// ```
+/// HTTPConfiguration(bindAddress: .interface(port: 443), tlsFilePaths: TLSFilePaths(certificatePath: "/some/path/cert.pem", keyPath: "/some/path/key.pem"))
+/// HTTPConfiguration(bindAddress: .address("localhost:8080"))
+/// ```
 public final class HTTPConfiguration: Configuration {
-    enum Defaults {
-        static let hostname = "0.0.0.0"
-        static let port = 8080
+    /// Default values for bindAddress
+    public enum Defaults {
+        public static let address = "localhost"
+        public static let httpPort = 80
+        public static let httpsPort = 443
+        public static let bindAddress = "0.0.0.0"
     }
     
-    enum HTTPConfigurationError: LocalizedError {
-        case incompatibleFlags
-
-        var errorDescription: String? {
-            switch self {
-            case .incompatibleFlags:
-                return "The command line arguments for HTTPConfiguration are invalid."
-            }
-        }
-
-        var recoverySuggestion: String? {
-            switch self {
-            case .incompatibleFlags:
-                return "Example usage of HTTPConfiguration: --hostname 0.0.0.0 --port 8080 or --bind 0.0.0.0:8080"
+    
+    public var bindAddress: BindAddress
+    public var hostname: Hostname
+    public var supportVersions: Set<HTTPVersionMajor> = [.one]
+    public var tlsConfiguration: TLSConfiguration?
+    
+    
+    public var uriPrefix: String {
+            let httpProtocol: String
+            var port = ""
+            
+            if self.tlsConfiguration == nil {
+                httpProtocol = "http://"
+                if hostname.port != 80 {
+                    port = ":\(hostname.port)"
+                }
+            } else {
+                httpProtocol = "https://"
+                if hostname.port != 443 {
+                    port = ":\(hostname.port)"
+                }
             }
```

Comment:
Please one level of indentation here.
```suggestion
    public var uriPrefix: String {
        let httpProtocol: String
        var port = ""
        
        if self.tlsConfiguration == nil {
            httpProtocol = "http://"
            if hostname.port != 80 {
                port = ":\(hostname.port)"
            }
        } else {
            httpProtocol = "https://"
            if hostname.port != 443 {
                port = ":\(hostname.port)"
            }
        }
```